Scripts/Python/xHighLevelStarTrekDoor.py

```diff
@@ -144,219 +118,76 @@ def OnSDLNotify(self,VARname,SDLname,playerID,tag):
-
-    ##########################################
-    def SendNote(self, ExtraInfo):
-        notify = ptNotify(self.key)
-        notify.clearReceivers()                
-        notify.addReceiver(self.key)        
-        notify.netPropagate(1)
-        notify.netForce(1)
-        notify.setActivate(1.0)
-        notify.addVarNumber(str(ExtraInfo),1.0)
-        notify.send()
+            if doorState == doorSDLstates['closetoopen']:
+                self.UpdateDoorState(doorSDLstates['opening'])
 
-    ##########################################
-    def UpdateRespStack (self):
-        #PtClearTimerCallbacks(self.key)
-        #Updates the Responder List
-        old = self.DoorStack.pop(0)
-        PtDebugPrint("xHighLevelStarTrekDoor: Getting rid of Resp: %s" % (old))
-        if len(self.DoorStack):
-            PtDebugPrint("xHighLevelStarTrekDoor: There's at lest one more Resp to play.")
-            code = self.DoorStack[0]            
-            PtDebugPrint("Playing command: %s" % (code))
-            #PtDebugPrint("xHighLevelStarTrekDoor: Timer set to : %d" % self.respondertime)
-            #PtAtTimeCallback(self.key,self.respondertime,1)
-            self.ExecCode(code)
-            if self.DoorStack[0].find('fastforward=1') != -1:
-                self.DoorStack.pop(0)
+            elif doorState in (doorSDLstates['movingclosed'], doorSDLstates['closing']):
+                self.UpdateDoorState(doorSDLstates['closed'])
 
     ##########################################
     def UpdateDoorState (self, StateNum):
-        if StateNum != self.DoorState:
+        if StateNum != self.SDL['DoorState'][0]:
             ageSDL = PtGetAgeSDL()
-            self.DoorState = StateNum
             self.SDL['DoorState'] = (StateNum,)
-            self.SendNote('DoorState='+str(StateNum))
-            PtClearTimerCallbacks(self.key)
-            PtAtTimeCallback(self.key, 30, 1)
             if self.DoorEnabled == 0:
-                return            
-
-            if self.DoorState == doorSDLstates['opening']:
-                self.SendNote("respOpenDoor;0")
-                PtDebugPrint("xHighLevelStarTrekDoor: Notifying Clients to play Open Door Responder")
-
-            elif self.DoorState == doorSDLstates['closing']:
-                self.SendNote("respCloseDoor;0")
-                PtDebugPrint("xHighLevelStarTrekDoor: Notifying Clients to play Close Door Responder")
-
-            elif self.DoorState == doorSDLstates['open']:
-                PtDebugPrint("xHighLevelStarTrekDoor: Updating Age SDL to Open")
+                return
+            if self.SDL['DoorState'][0] == doorSDLstates['opening']:
                 ageSDL[strDoorClosedVar.value] = (0,)
-
-            elif self.DoorState == doorSDLstates['closed']:
-                PtDebugPrint("xHighLevelStarTrekDoor: Updating Age SDL to Closed")
+            elif self.SDL['DoorState'][0] == doorSDLstates['closing']:
                 ageSDL[strDoorClosedVar.value] = (1,)
```

Comment:
This could be improved greatly by making the door state into a [property](https://www.freecodecamp.org/news/python-property-decorator/).